src/palo_alto_firewall_analyzer/core.py

```diff
@@ -178,11 +184,34 @@ def cached_fqdn_lookup(domain):
 
 
 @functools.lru_cache(maxsize=None)
-def xml_object_to_dict(xml_obj):
+def xml_object_to_dict1(xml_obj):
     obj_xml_string = xml.etree.ElementTree.tostring(xml_obj)
     obj_dict = xmltodict.parse(obj_xml_string)
     return obj_dict
 
+def xml_object_to_dict(xml_obj):
+    obj_xml_string = xml.etree.ElementTree.tostring(xml_obj) 
+    
+    root = xml.etree.ElementTree.fromstring(obj_xml_string)    
+    
+    list_atr_remove = ['loc']
```

Comment:
Could you add a comment here as to why you're removing the 'loc' attribute

Reply:
Okay, I'll add a comment, but I'll also comment it here just in case you see another solution. The issue is that they've added the 'loc' attribute, and now the script doesn't retrieve the data properly from the XML. Here's an example.

![image](https://github.com/moshekaplan/palo_alto_firewall_analyzer/assets/26645553/e2302e7f-9267-4189-89e8-2ccb58928936)


What I've done is clean the XML of attributes. I did it this way in case they add other attributes later on.

Reply:
Thanks for explaining

Reply:
Sorry for the long response time, my work and personal lives have both been very busy.

I was doing some testing and noticed a problem:
Some validators retrieve attributes, like the name, as is done here: https://github.com/moshekaplan/palo_alto_firewall_analyzer/blob/main/src/palo_alto_firewall_analyzer/validators/misleading_objects.py#L27 :

```
address_dict = xml_object_to_dict(address_entry)
entry_name = address_dict['entry']['@name']
```

Therefore, removing the attributes breaks at least that validator - probably more.

Reply:
Hi, no problem at all, I understand this is a hobby, there are more important things. I don't understand what you mean, I've been testing and I wasn't getting any errors, well, except for my files,... but the only tag I remove is 'loc' and I make it nominal, I don't remove any other name or anything. Could you provide some kind of example of your tests? As I mentioned, with the exported files I've been working with, both with the '@loc' tag and without it, I haven't had any issues. Thanks.

Reply:
Sorry for the super-delayed follow-up on this.

Say I have an address object with a single IP-netmask of 127.0.0.1. 

The old code (now called with `xml_object_to_dict1`) would result in a dict like the following:
> (Pdb) xml_object_to_dict1(address_entry)
{'entry': {'@name': 'ip-127.0.0.1', 'ip-netmask': '127.0.0.1'}}

The new code (what in this PR is called `xml_object_to_dict`) removes the `@name` attribute, resulting in the following:
> (Pdb) xml_object_to_dict(address_entry)
{'entry': {'ip-netmask': '127.0.0.1'}}

This is problematic because of code that accesses the `@name` attribute, like:
https://github.com/Nawtest/palo_alto_firewall_analyzer/blob/main/src/palo_alto_firewall_analyzer/validators/misleading_objects.py#L28 :
> entry_name = address_dict['entry']['@name']

Because the `@name` attribute is missing, the dictionary access fails

Reply:
Okiz, don't worry, I'll review it. It's strange to me that this attribute gets removed; the one I remove is '@loc'. I'll check it to fix it.

Reply:
Hi, I am checking the code, and I don't understand why removing the 'loc' tag causes an error when looking for the 'name' tag. It is true that in my XML files, there are some entries without the 'name' tag. I added an if statement to check the issue, and this is the result:
- if not delete the 'loc' tag, member is:
![image](https://github.com/user-attachments/assets/cec3cca0-a891-45a3-aeb2-b7b5d23acf73)

- if delete the 'loc' tag, and include a if, the json generated is:
![image](https://github.com/user-attachments/assets/b99b571e-72ab-4bf3-8b83-54ebaa35c13e)

This example my file:
![image](https://github.com/user-attachments/assets/3b827a19-55cd-4b7e-9c00-d7ff48ff3756)

Not include 'name' tag